northstar/src/runtime/process/mod.rs

```diff
@@ -322,8 +323,31 @@ impl super::state::Process for Process {
                 let process_group = unistd::Pid::from_raw(-(self.pid as i32));
                 let sigkill = Some(sys::signal::SIGKILL);
                 sys::signal::kill(process_group, sigkill).ok();
+                return Ok(());
             }
-        }
+        };
+
+        // spawn a task that waits for the process exit status
+        let tx = self.tx.clone();
```

Comment:
What about moving `tx` to the args of `spawn`?

Reply:
do you mean making `self.tx` and `Option` and then `take()` it here?

Reply:
No. I mean

```rust
async fn spawn(&mut self, tx: EventTx) -> Result<(), Error> {
```

because after the `spawn`/`start` sequence `Process::tx` is never used again.

Reply:
so changing the `Process` trait, I guess it's possible but isn't it a bit weird from the interface point of view? I agree that there is some smell here, I think it originates from the channel ownership and who should take care of notify when the process exits, but perhaps we should address that in the next iteration.